tests/test_simpleButler.py

```diff
@@ -767,5 +757,68 @@ def test_skypix_templates(self):
         self.assertEqual(path, "test/warp/HSCA02713600_12345_12345_12345_12345")
 
 
+class DirectSimpleButlerTestCase(SimpleButlerTests, unittest.TestCase):
+    """Run tests against DirectButler implementation."""
+
+    datasetsManager = (
+        "lsst.daf.butler.registry.datasets.byDimensions.ByDimensionsDatasetRecordStorageManagerUUID"
+    )
+
+    def setUp(self):
+        self.root = makeTestTempDir(TESTDIR)
+
+    def tearDown(self):
+        removeTestTempDir(self.root)
+
+    def makeButler(self, writeable: bool = False) -> Butler:
+        config = ButlerConfig()
+
+        # make separate temporary directory for registry of this instance
+        tmpdir = tempfile.mkdtemp(dir=self.root)
+        config["registry", "db"] = f"sqlite:///{tmpdir}/gen3.sqlite3"
+        config["registry", "managers", "datasets"] = self.datasetsManager
+        config["root"] = self.root
+
+        # have to make a registry first
+        registryConfig = RegistryConfig(config.get("registry"))
+        _RegistryFactory(registryConfig).create_from_config()
+
+        butler = Butler.from_config(config, writeable=writeable)
+        DatastoreMock.apply(butler)
+        return butler
+
+
+@unittest.skipIf(create_test_server is None, "Server dependencies not installed.")
+class RemoteSimpleButlerTestCase(SimpleButlerTests, unittest.TestCase):
+    """Run tests against Butler client/server."""
+
+    supportsCollectionRegex = False
+
+    def makeButler(self, writeable: bool = False) -> Butler:
+        server_instance = self.enterContext(create_test_server(TESTDIR))
+        butler = server_instance.hybrid_butler
+        DatastoreMock.apply(butler)
```

Comment:
Maybe add a comment here that the mock can't be used directly because remote butler isn't using a real datastore interface?